### PR TITLE
Warden Auth

### PR DESCRIPTION
**Pull request by joshuashort**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -47,29 +49,38 @@
     "gatsby-transformer-sharp": "^3.4.0",
     "hcaptcha": "^0.1.0",
     "isomorphic-dompurify": "^0.19.0",
-    "jsonwebtoken": "^8.5.1",
     "mailgun.js": "^6.0.1",
+    "https-browserify": "^1.0.0",
+    "isomorphic-dompurify": "^0.19.0",
```
Comment (Simon-Busch) [inline package.json:54]: ! duplicate of "isomorphic-dompurify"
```suggestion
```

File: src/components/Login/Login.tsx [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+          href=""
+          onClick={(e) => handleLogin(e, "walletConnect")}
+          className={styles.Link}
+        >
+          <img
+            src="/images/wallet-connect-logo.svg"
+            alt="logout icon"
+            className={styles.Icon}
+          />
+          Connect WalletConnect
```
Comment (Simon-Busch) [inline src/components/Login/Login.tsx:123]: Missing image in /static/images

File: src/components/Login/Login.tsx [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+      </Dropdown>
+      <div className={styles.Mobile}>
+        <a href="" onClick={(e) => handleLogin(e)} className={styles.Link}>
+          <img
+            src="/images/meta-mask-logo.svg"
+            alt="logout icon"
+            className={styles.Icon}
+          />
+          Connect MetaMask
+        </a>
```
Comment (Simon-Busch) [inline src/components/Login/Login.tsx:112]: Missing image in /static/images

File: src/components/Login/Login.tsx [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+        <button
+          type="button"
+          onClick={(e) => handleLogin(e, "walletConnect")}
+          className={clsx(dropdownStyles.Button, styles.Desktop)}
+        >
+          <img
+            src="/images/wallet-connect-logo.svg"
+            alt="logout icon"
+            className={styles.Icon}
+          />
```
Comment (Simon-Busch) [inline src/components/Login/Login.tsx:100]: Missing image in /static/images

**Review by Simon-Busch — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (Simon-Busch) [reply package.json]: Done in my commit [faefe57](https://github.com/code-423n4/code423n4.com/pull/1186/commits/faefe57a9be65bc7a97aec57062757e5a9f410ab)

File: src/components/Login/Login.tsx (hunk 3 above)
Comment (Simon-Busch) [reply src/components/Login/Login.tsx]: added in https://github.com/code-423n4/code423n4.com/pull/2047

File: src/components/Login/Login.tsx (hunk 4 above)
Comment (Simon-Busch) [reply src/components/Login/Login.tsx]: added in https://github.com/code-423n4/code423n4.com/pull/2047

File: functions/register-warden.js [hunk 5]
```diff
@@ -72,7 +127,52 @@ exports.handler = async (event) => {
...
+    // make sure that an update has a valid warden file, and a new registration does not
+    const response = await fetch(url);
+    if (isUpdate && !response.ok) {
+      return {
+        statusCode: 422,
+        body: "Account does not exist",
+      };
+    }
+    const userData = await response.json();
+    if ((!isUpdate && response.ok) || userData.moralisId) {
```
Comment (captainmangoC4) [inline functions/register-warden.js:140]:
```suggestion
    if (!isUpdate && (response.ok || userData.moralisId)) {
```

File: functions/register-warden.js [hunk 6]
```diff
@@ -99,28 +199,58 @@ exports.handler = async (event) => {
...
+        );
+        files[`_data/handles/${handle}.json`] = content;
+      } catch (error) {
+        console.error(error);
+        return {
+          statusCode: 500,
+          body: JSON.stringify({ error: "Internal server error." }),
+        };
+      }
+    }
```
Comment (captainmangoC4) [inline functions/register-warden.js:230]:
```suggestion
        const content = JSON.stringify(
          {
            ...userData,
            moralisId,
          },
          null,
          2
        );
        files[`_data/handles/${handle}.json`] = content;
    }
```

Comment (captainmangoC4) [reply functions/register-warden.js:230]: user data is already fetched on line 132

File: .gitignore [hunk 7]
```diff
@@ -1,8 +1,12 @@
 node_modules
 .DS_Store
 public
+.netlify
```
Comment (joshuashort) [inline .gitignore:4]: dup.

```suggestion
```

**Review by Simon-Busch — APPROVED** [review]

File: _data/handles/CaptainMango.json [hunk 8]
```diff
@@ -0,0 +1,6 @@
+{
+  "handle": "CaptainMango",
+  "link": "https://github.com/captainmangoC4",
+  "moralisId": "E4nTmnd02Q3BoI9s9rRHFr7L",
+  "image": "./avatars/CaptainMango.jpeg"
+}
```
[the remark this answers is not in the document]
Comment (captainmangoC4) [reply _data/handles/CaptainMango.json]: I think it's fine to leave them in. We will be able to use them in the future for other testing purposes. A few other team members have warden accounts as well.

**Comment by sockdrawermoney** [comment]
🎉

**Comment by Simon-Busch** [comment]
As discussed with @captainmangoC4 we will re-work the ability and impact of updating the C4 username after the launch.

reference to [7d53f2f](https://github.com/code-423n4/code423n4.com/pull/1186/commits/7d53f2fbc2a6a65082327efe5edfe066f78d5996)

**Pull request merged**